crates/xtask-changes/src/xtask.rs

```diff
@@ -0,0 +1,432 @@
+                    std::task::Poll::Ready(res) => {
+                        break res?;
+                    }
+                    std::task::Poll::Pending => registry.block_until_ready()?,
+                }
+            };
+            let published = possibilities.iter().map(|s| s.version()).max().cloned();
+
+            results.push(ChangedPackage {
+                name,
+                local,
+                published,
+                commits,
+            });
+        }
+    }
+
+    for changed in &results {
+        let name = &changed.name;
+        let local = &changed.local;
+        let published = changed
+            .published
+            .as_ref()
+            .map(|p| p.to_string())
+            .unwrap_or_else(|| "-".to_owned());
+        let _ = config.shell().status(
+            "Changed",
+            format!("{name} {local} was last published as {published}"),
+        );
+        let prefix = format!("{:>13}", " ");
+        for commit in &commits {
+            let _ = config.shell().write_stderr(&prefix, &ColorSpec::new());
+            let _ = config.shell().write_stderr(
+                &commit.short_id,
+                ColorSpec::new().set_fg(Some(Color::Yellow)),
+            );
+            let _ = config.shell().write_stderr(" ", &ColorSpec::new());
+            let _ = config
+                .shell()
+                .write_stderr(&commit.summary, &ColorSpec::new());
+
+            let current_status = commit.status();
+            write_status(current_status, config);
+            let _ = config.shell().write_stderr("\n", &ColorSpec::new());
+        }
+    }
+
+    Ok(())
+}
+
+fn write_status(status: Option<CommitStatus>, config: &cargo::util::Config) {
+    if let Some(status) = status {
+        let suffix;
+        let mut color = ColorSpec::new();
+        match status {
+            CommitStatus::Breaking => {
+                suffix = format!(" ({})", status);
+                color.set_fg(Some(Color::Red));
+            }
+            CommitStatus::Feature => {
+                suffix = format!(" ({})", status);
+                color.set_fg(Some(Color::Yellow));
+            }
+            CommitStatus::Fix => {
+                suffix = format!(" ({})", status);
+                color.set_fg(Some(Color::Green));
+            }
+            CommitStatus::Ignore => {
+                suffix = String::new();
+            }
+        }
+        let _ = config.shell().write_stderr(suffix, &color);
+    }
+}
+
+#[derive(Clone, Debug, PartialOrd, Ord, PartialEq, Eq)]
+pub struct ChangedPackage {
+    name: InternedString,
+    local: semver::Version,
+    published: Option<semver::Version>,
+    commits: Vec<CommitDescription>,
+}
+
+#[derive(Clone, Debug, PartialOrd, Ord, PartialEq, Eq)]
+pub struct CommitDescription {
+    id: git2::Oid,
+    short_id: String,
+    summary: String,
+    message: String,
+    paths: std::collections::BTreeSet<std::path::PathBuf>,
+}
+
+impl CommitDescription {
```

Comment:
I would probably suggest to remove conventional commit part. Let's focus on the original goal for now.